src/client/Microsoft.Identity.Client.Broker/RuntimeBroker.cs

```diff
@@ -41,8 +41,9 @@ internal class RuntimeBroker : IBroker
 
                 return new NativeInterop.Core();
             }
-            catch (Exception ex)
+            catch (MsalRuntimeException ex) when (ex.Status == ResponseStatus.ApiContractViolation)
```

Comment:
So this would catch if MSAL fails to load runtime DLL? I guess the `ApiContractViolation` status is a bit misleading in this situation.

In what other error scenarios would this status appear?

What is end-to-end developer experience here if runtime DLL is missing? So we call IsBrokerInstalledAndInvokable, which returns falls. If OS is unsupported, I think we give descriptive error, right? But what if some package restore failed, how would the user know? Will the user be able to self help in that scenario?

Reply:
right, so based on the testing with the changes I did earlier here are the findings. 

`With catch (MsalRuntimeException ex) when (ex.Status == ResponseStatus.ApiContractViolation)`

This will happen on older versions of Windows. Which means broker is not available. But, We will never get here with our current OS version check. Instead in this scenario we will fallback to the browser. MSALRuntime does it's internal check for OS compatibility and throws an ApiContractViolation MsalRuntimeException. if, For any reason, if our OS check fails then this will catch the MsalRuntimeException and log but we will not fallback to the browser.

**for the actual fix we need this line,** 

```
catch (Exception ex)
{
    // When MSAL Runtime dlls fails to load then we catch the exception and throw with a meaningful
    // message with information on how to troubleshoot
    throw new MsalClientException(
        "wam_runtime_init_failed", ex.Message + " See https://aka.ms/msal-net-wam#troubleshooting");
}
```
This catches the dll load exception and throws with instructions on how to fix the issue.

Reply:
@bgavrilMS please review